docs/how-tos/use-multi-format-reader.md

```diff
@@ -0,0 +1,399 @@
+    "/ENTRY[entry]/SAMPLE[sample]/physical_form": "powder",
+    "/ENTRY[entry]/SAMPLE[sample]/temperature": 300,
+    "/ENTRY[entry]/SAMPLE[sample]/temperature/@units": "K"
+}
+```
+Note that here we are using `parent_key="/ENTRY[entry]"` as well as a `CONVERT_DICT`, meaning that each key in `self.eln_data` will start with `"/ENTRY[entry]"` and some of the paths will be converted to match the template notation. This will be important later.
+
+## Parsing the config file
+Next up, we can make us of the config file, which is a JSON file that tells the reader how to map the concepts from the HDF5 and ELN files in order to populate the template designed to match `NXsimple`. Essentially, the config file should contain all keys that are present in the NXDL. In our case, the config file looks like this:
+
+I recommend to make the statement even stronger, the choices made in the config file define how semantics from the src and trg (NeXus) side are mapped. 
+
+
+```json
+{
+  "/ENTRY/title": "@eln", 
+  "/ENTRY/USER[user]": {
+    "name":"@eln",
+    "address":"@eln"
+  }, 
+  "/ENTRY/INSTRUMENT[instrument]": {
+    "@version":"@attrs:metadata/instrument/version",
+    "DETECTOR[detector]":{
+      "count_time":"@attrs:metadata/instrument/detector/count_time",
+      "count_time/@units":"@attrs:metadata/instrument/detector/count_time_units"
+    }
+  },
+  "/ENTRY/SAMPLE[sample]": {
+    "name":"@eln",
+    "physical_form":"@eln",
+    "temperature":"@eln",
+    "temperature/@units":"@eln"
+  },
+  "/ENTRY/data": {
+    "@axes":["x_values"],
+    "@signal": "data",
+    "data": "@data:y_values",
+    "x_values": "@data:x_values"
+  }
+}
+```
+We also implement a method for setting the config file in the reader:
+```python
+def set_config_file(self, file_path: str) -> Dict[str, Any]:
+    if self.config_file is not None:
+        logger.info(
+            f"Config file already set. Replaced by the new file {file_path}."
+        )
+    self.config_file = file_path
+  
+    return {}
+```        
+
+## Filling the template from the read-in data
+Finally, after reading in all of the data and metadata as well as designing the config file, we can start filling the template. For this, we must implement functions that are called using the reader's **callbacks**.
+
+We will start with the `@attrs` prefix, associated with the `attrs_callback`. We must implement the `get_attr` mehthod:
+```python
+def get_attr(self, key: str, path: str) -> Any:
+    """
+    Get the metadata that was stored in the main file.
+    """
+    if self.hdf5_data is None:
+        return None
+    
+    return self.hdf5_data.get(path)
+```
+Here, we are making use of the `path`, which is the part of the config value that comes after the `@attrs:` prefix. For example, for the config value `"@attrs:metadata/instrument/version"`, the extracted path is `metadata/instrument/version`, which is also one of the keys of the `self.hdf5_data` dictionary.

```

Comment:
Also mention keys and what it contains

Reply:
I added this in the ELN part where we actually use the `key`.

Reply:
EDIT: added a longer description what `key` and `path` are above.